### PR TITLE
An attempt to fix our failing builds

### PR DESCRIPTION
**Pull request by jakule**
Add checkout out action to fix `Error: The process '/usr/bin/git' failed with exit code 128` as mentioned in 
https://github.com/dorny/paths-filter/issues/60

### CODE REVIEW

File: .github/workflows/aws-e2e-tests-non-root.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (zmb3) [inline .github/workflows/aws-e2e-tests-non-root.yaml:32]: It looks like the checkout is only necessary on the `merge_group` trigger, not on `pull_request`. Can we filter this out with an `if`?

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: .github/workflows/aws-e2e-tests-non-root.yaml [hunk 2]
```diff
@@ -29,6 +29,9 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (reedloden) [inline .github/workflows/aws-e2e-tests-non-root.yaml:34]:
```suggestion
         uses: actions/checkout@v4
```

File: .github/workflows/check-devbox.yaml [hunk 3]
```diff
@@ -15,6 +15,9 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (reedloden) [inline .github/workflows/check-devbox.yaml:20]:
```suggestion
         uses: actions/checkout@v4
```

File: .github/workflows/doc-tests.yaml [hunk 4]
```diff
@@ -15,6 +15,9 @@ jobs:
     outputs:
       changed: ${{ steps.changes.outputs.changed }}
     steps:
+      - name: Checkout
+        if: ${{ github.event_name == 'merge_group' }}
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (reedloden) [inline .github/workflows/doc-tests.yaml:20]:
```suggestion
         uses: actions/checkout@v4
```

File: .github/workflows/integration-tests-non-root.yaml [hunk 5]
```diff
@@ -15,6 +15,9 @@ jobs:
     outputs:
       changed: ${{ steps.changes.outputs.changed }}
     steps:
+      - name: Checkout
+        if: ${{ github.event_name == 'merge_group' }}
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (reedloden) [inline .github/workflows/integration-tests-non-root.yaml:20]:
```suggestion
         uses: actions/checkout@v4
```

File: .github/workflows/unit-tests-integrations.yaml [hunk 6]
```diff
@@ -15,6 +15,9 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
Comment (reedloden) [inline .github/workflows/unit-tests-integrations.yaml:20]:
```suggestion
         uses: actions/checkout@v4
```

**Review by reedloden — COMMENTED** [review]
Please use `actions/checkout@v4` instead of specific commit hash. We permit rolling tags for GitHub-blessed actions under `actions/*`.

**Review by reedloden — APPROVED** [review]

**Review by espadolini — APPROVED** [review]

**Pull request merged**